app/services/script-tags.ts

```diff
@@ -0,0 +1,113 @@
+/* eslint-disable no-mixed-spaces-and-tabs*/
+/* eslint-disable object-shorthand*/
+import Service, { inject as service } from '@ember/service';
+import config from 'ember-get-config';
+import Intl from 'ember-intl/services/intl';
+import CurrentUserService from 'ember-osf-web/services/current-user';
+import { MetaTagAttrs } from 'ember-osf-web/services/meta-tags';
+import captureException from 'ember-osf-web/utils/capture-exception';
+
+export type Content = object | string | String | number | null | undefined;
+
+export type DataContent = Content | Content[];
+
+export interface ScriptTagsData {
+    type?: DataContent;
+    content?: DataContent;
+}
+
+export interface ScriptTagDef {
+    [s: string]: DataContent;
+}
+
+export interface JSONLDScriptTagAttrs {
+    type: Content;
+}
+
+export type ScriptTagAttrs = JSONLDScriptTagAttrs;
+
+// ember-cli-meta-tags element types
+export enum TagType {
+    BASE = 'base',
+    LINK = 'link',
+    META = 'meta',
+    SCRIPT = 'script',
+    NOSCRIPT = 'noscript',
+    TITLE = 'title',
+}
+
+export interface HeadTagDef {
+    type: string;
+    content: DataContent;
+    attrs: MetaTagAttrs | ScriptTagAttrs;
+}
+
+/**
+ *
+ * Creates one script tag with model-specific overrides or default attribute values.
+ * Should additional script tags be needed, getHeadTags() should be called for each
+ * script and added to the headTags array.
+ *
+ * @method getScriptTagAttributes
+ * @method getHeadTags
+ * @param {ScriptTagsData} scriptTagsOverrides Overrides defaults
+ * @return {HeadTagsDefs} getHeadTags() returns HeadTagDefs for ScriptTagAttrs
+ */
+export default class ScriptTags extends Service {
+    @service intl!: Intl;
+    @service router!: any;
+    @service currentUser!: CurrentUserService;
+
+    async returnStructuredData(guid: string): Promise<any> {
+        const url = `${config.OSF.url}/${guid}/metadata/?format=google-dataset-json-ld`;
+        let jsonLD = {};
+        let jsonFetch: {} | void;
+        try {
+            jsonFetch = await this.returnJSON(url);
+            if (jsonFetch && (typeof(jsonFetch) === 'object')) {
+                jsonLD = jsonFetch;
+            }
+        } catch (e) {
+            const errorMessage = this.intl.t('general.structured_data.json_ld_retrieval_error');
+            captureException(e, { errorMessage });
+        }
+        return jsonLD;
+    }
+
+    async returnJSON(url: string) {
+        const ajax = await this.currentUser.authenticatedAJAX({
+            method: 'GET',
+            url: url,
+        });
+        return ajax;
+    }
```

Comment:
We resolved this during our conversation, and am resolving it here as well.

Reply:
IIRC, you mentioned that nested function calls are needed because `authenticatedAJAX` returns a promise. But in fact it is not. Simply `await this.currentUser.authenticatedAJAX` would do.